platforms/chibios/drivers/spi_master.c

```diff
@@ -259,6 +306,10 @@ bool spi_start(pin_t slavePin, bool lsbFirst, uint8_t mode, uint16_t divisor) {
 }
 
 spi_status_t spi_write(uint8_t data) {
+#if defined(QMK_MCU_SERIES_STM32H7XX)
+    SCB_CleanDCache_by_Addr((uint32_t *)(((uint32_t)&data) & ~(uint32_t)0x1F), 1 + 32);
+#endif
+
     uint8_t rxData;
     spiExchange(&SPI_DRIVER, 1, &data, &rxData);
```

Comment:
Technically this also breaks cache coherency for `rxData`, but we also don't care about that data…

One potentially problematic case could be if some unrelated code reads a stale value from the cache, then does something else that causes that cache line to be evicted (without writing it to RAM, because the cache line was not marked as modified), and then reads from that address again, getting a different value this time.  However, with code like this such behavior could arise only if the code reads some uninitialized data.  Still, the proper solution would be to flush the cache appropriately, and probably even move the buffer to a separate cache line (this would also remove the case of DMA to stack).